specs/language/basic.tex

```diff
@@ -1,5 +1,112 @@
 \Ch{Basic Concepts}{Basic}
 
+\Sec{Types}{Basic.types}
+
+\p The \textit{object representation} of an object of type \texttt{T} is the
+sequence of \textit{N} bytes taken up by the object of type \texttt{T}, where
+\textit{N} equals \texttt{sizeof(T)}\footnote{\texttt{sizeof(T)}eturns the size
+of the struct as-if it's stored in device memory, and determining the size if
```

Comment:
Is there a glossary entry for "device memory"?  I imagine by this term you mean something like host-visible device memory or some specific memory type that is a subset of this.  Non-host-visible device memory layouts (including group shared memory, local memory for registers/arrays, inter-stage attributes, inter-function parameters/payloads, etc...) could be optimized in ways that are not visible to the host application or directly observable by the shader.

Reply:
"device memory" is defined in `Intro.Memory.Spaces` which is earlier in the document. The end of the next sentence has a reference to that section of the document.

Reply:
I think the definition of `sizeof` is particularly problematic and we don't have a well enough defined memory model to cover all the relevant cases.

Reply:
I'm going to mark this conversation as resolved since I don't know that we have an actionable change here. I think we have some bigger issues around clearly defining the memory model for HLSL (and DXIL) that we'll also need to figure out over the coming months. We'll also need to revisit this wording as we try to close in on a better definition for `sizeof`.